rxtools/source/lib/fatfs/diskio.c

```diff
@@ -56,14 +56,14 @@ DRESULT disk_read (
 {
     switch(pdrv){
         case 0:
-            if (sdmmc_sdcard_readsectors(sector,count,buff))
+            if (sdmmc_sdcard_readsectors(sector,count,(uint8_t *)buff))
                 return RES_PARERR;
             break;
         case 1:
-            nand_readsectors(sector, count, buff, CTRNAND);
+            nand_readsectors(sector, count, (uint8_t *)buff, CTRNAND);
             break;
         case 2:
-            emunand_readsectors(sector, count, buff, CTRNAND);
+            emunand_readsectors(sector, count, (uint8_t *)buff, CTRNAND);
```

Comment:
Well good. if so you might make it a void*. To get rid of unsigned/signed or types.